### PR TITLE
Add IOT central app templates api in swagger

### PR DESCRIPTION
**Pull request by kawayiboy**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5603 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5603/commits/c51c6b5de90c4c3a04206c233ffe8994efad3c22

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json [hunk 1]
```diff
@@ -410,6 +410,43 @@
...
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The body contains all the non-security properties of the IoT Central application templates. Security-related properties are set to null.",
```
Comment (emgarten) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: What are security properties here

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/examples/Apps_Templates.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "api-version": "2018-09-01"
+  },
+  "responses": {
+    "200": {
+      "description": "Returns the application templates details",
+      "headers": {
+        "code": "200"
+      },
+      "body": {
```
Comment (r-delgadillo) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/examples/Apps_Templates.json]: Looka t the other examples. I don't see why `id`, `location`, and other properties listed below would be in the response body for getting application templates.

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json [hunk 3]
```diff
@@ -410,6 +410,43 @@
         "deprecated": false
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.IoTCentral/appTemplates": {
```
Comment (r-delgadillo) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: We decided to make this global?

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json [hunk 4]
```diff
@@ -410,6 +410,43 @@
...
+      "get": {
+        "tags": [
+          "Apps"
+        ],
+        "description": "Get all available application templates.",
+        "operationId": "Apps_Templates",
```
Comment (r-delgadillo) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: nit: "App_Templates"

Comment (r-delgadillo) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: Same for file naming

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json [hunk 5]
```diff
@@ -733,6 +770,60 @@
...
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The name of the template.",
+          "type": "string",
+          "readOnly": true
+        },
+        "title": {
+          "description": "The localized title of the template.",
```
Comment (r-delgadillo) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: We don't have to specify it's localized, do we?

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json [hunk 6]
```diff
@@ -733,6 +770,60 @@
...
+          "readOnly": true
+        },
+        "description": {
+          "description": "The description of the template.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "AppTemplates": {
```
Comment (r-delgadillo) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: nit: looks like our naming convention should be `AppTemplateResult` or something like that. Having `AppTemplate` and `AppTemplates` could be a bit confusing.

Comment (r-delgadillo) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: nit: looks like our naming convention should be `AppTemplateResult` or something like that. Having `AppTemplate` and `AppTemplates` could be a bit confusing.

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json (hunk 4 above)
Comment (kawayiboy) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: yes, but I'm trying to follow other naming convention, Apps_xxx

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json (hunk 3 above)
Comment (kawayiboy) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: we would firstly go with global, considering the customers can do separate CRUD operations on the templates

Comment (r-delgadillo) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: We could also do crud if it's regional.

**Pull request closed without merging**